pid.py

```diff
@@ -1,13 +1,42 @@
 class PID:
-	def __init__(self):
+	def __init__(self, a, b, c, initial_error=None):
+		# (a, b, c) are the weights for (p, i, d)
+		self.a = a
+		self.b = b
+		self.c = c
+
+		# Proportional, Integral, & Derivative
 		self.p = 0
 		self.i = 0
 		self.d = 0
 
-	def update(self, target_alt, alt, vel, acc, time_steps):
-		self.p = 0.4 * (target_alt - alt)
-		self.i += 0.001 * (target_alt - alt) / time_steps
-		self.d = -2 * vel
-	
-	def get_value(self):
+		# Provided an initial error, self.prepare isn't necessary
+		if initial_error is not None:
+			self.last_error = initial_error
+
+	def __call__(self, *args):
+		self.update(*args)
+		return self.output()
+		
+	def prepare(self, current, target):
+		""" Run at start to give the controller an initial error
+			from which to calculate the first derivative
+		"""
+		self.last_error = current - target
```

Comment:
I think this should take in an error as an argument instead of it calculating the error on its own.